### PR TITLE
Keep quota history by archiving

### PR DESCRIPTION
**Pull request by cben**
Depends on:
- [x] https://github.com/ManageIQ/manageiq/pull/17020 — merged
- [x] https://github.com/ManageIQ/more_core_extensions/pull/59 — released 3.6.0, bumped by this PR
- [x] https://github.com/ManageIQ/manageiq/pull/16722 

https://bugzilla.redhat.com/show_bug.cgi?id=1504560
@miq-bot add-label enhancement, gaprindashvili/yes

### CODE REVIEW

**Comment by cben** [comment]
- [ ] This archives and recreates all quota items each refresh.  Maybe graph refresh needs fix similar to https://github.com/ManageIQ/manageiq/pull/16723.

**Comment by cben** [comment]
These are ready, only [WIP] due to dependencies.  @zeari @yaacov please review.

**Comment by cben** [comment]
@zeari @yaacov @agrare deps merged, please review this & https://github.com/ManageIQ/manageiq/pull/16722.

File: app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+require 'bigdecimal'
```
Comment (zeari) [inline app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb]: This makes me very happy :1st_place_medal:

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit ca32b8981e6b7aee82177e1ff39673136535aa8a
Author: Adam Grare <agrare@redhat.com>
Date:   Tue Mar 13 15:53:05 2018 -0400

    Merge pull request #198 from cben/quota-history
    
    Keep quota history by archiving
    (cherry picked from commit 5ec31943f85dcbdbc1f37026430019bffcc6ed60)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1559544
```

**Pull request merged**